File: Advanced Programming Methods/toyLanguage/toyLanguage/model/ConstExp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace toyLanguage
{
    class ConstExp : Exp
    {
        private int number;

        public ConstExp(int n)
        {
            number = n;
        }

        public int Number
        {
            get { return number; }
            set { number = value; }
        }

        public override int Eval(IMyDictionary<string, int> sym)
        {
            return number;
        }

        public override String ToString()
        {
            return number.ToString();
        }

    }
}

File: Advanced Programming Methods/toyLanguage- Serializable/src/sample/model/ConstExp.java
package sample.model;

import sample.utils.MyIDictionary;
import sample.utils.MyIHeap;

public class ConstExp extends Exp {  //2
    private int number;
    public ConstExp(int number){
        this.number=number;
    }
    @Override
    public int eval(MyIDictionary<String, Integer> symbolTable, MyIHeap<Integer, Integer> hp){
        return number;
    }

    @Override
    public String toString(){
        return ""+this.number;
    }
}
